exercises/dominoes/src/test/java/DominoesTest.java

```diff
@@ -0,0 +1,228 @@
+import org.junit.Ignore;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+import org.junit.Rule;
+import org.junit.rules.ExpectedException;
+
+public class DominoesTest {
+	
+	@Rule
+	public ExpectedException expectedException = ExpectedException.none(); 
+	
+	@Test
+	public void emtpyInputEmptyOutputTest() {
+		Dominoes dominoes = new Dominoes();
+		
+		ArrayList<Domino> dominoesArray = new ArrayList<>();
+		
+		List<Domino> chain = dominoes.formChain(dominoesArray);
+		
+		assertEquals("The output list should be empty.", 0, chain.size());
+	}
+	
+	@Ignore("Remove to run test")
+	@Test
+	public void singletonInputSingletonOutput() {
+		Dominoes dominoes = new Dominoes();
+		
+		Domino[] dominoesList = {new Domino(1, 1)};
+		ArrayList<Domino> dominoesArray = new ArrayList<>(Arrays.asList(dominoesList));
```

Comment:
You don't need to make this an `ArrayList` anymore which should simplify this code a bit 🙂